pkg/migrations/sql/idempotent/base.sql

```diff
@@ -740,6 +740,33 @@ COMMENT ON FUNCTION SCHEMA_CATALOG.get_or_create_label_array(text, text[], text[
 IS 'converts a metric name, array of keys, and array of values to a label array';
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.get_or_create_label_array(TEXT, text[], text[]) TO prom_writer;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_or_create_label_ids(metric_name TEXT, label_keys text[], label_values text[])
+RETURNS TABLE(pos int, id int, label_key text, label_value text) AS $$
+        SELECT
+            -- only call the functions to create new key positions
+            -- and label ids if they don't exist (for performance reasons)
+            coalesce(lkp.pos,
+              SCHEMA_CATALOG.get_or_create_label_key_pos(get_or_create_label_ids.metric_name, kv.key)) idx,
+            coalesce(l.id,
+              SCHEMA_CATALOG.get_or_create_label_id(kv.key, kv.value)) val,
```

Comment:
Haven't gone into details, but is 749 and 751 dependent on each other? (if yes, then I hope they are in proper sync).

Reply:
There should be no dependency here